src/price_data.py

```diff
@@ -326,3 +328,133 @@ def get_cost(self, tr: Union[Operation, SoldCoin], reference_coin: str = config.
         if isinstance(tr, SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+    
+    def get_candles(self, start: int, stop: int, symbol: str) ->list:
+        if self.exchange.has['fetchOHLCV']:
+            sleep(self.exchange.rateLimit / 1000)  # time.sleep wants seconds
+            # get 2min before and after range
+            return self.exchange.fetch_ohlcv(symbol, '1m', start-1000*60*2, max(int((stop-start)/1000/60)+5, 1))
+        else:
+            logging.warning(
+                "fetchOHLCV not implemented on exchange, skipping priceloading using ohlcv")
+            raise None
+
+    def initialize_ccxt(self):
+        exchange_id = 'binance'
+        exchange_class = getattr(ccxt, exchange_id)
+        self.exchange = exchange_class()
+        self.markets = []
+        markets = self.exchange.fetch_markets()
+
+        for market in markets:
+            # may not apply for all exchanges, currently works for binance
+            # caches a list of all pairs on the exchange
+            self.markets.append(market["symbol"].split("/"))
```

Comment:
yes that would be nice. I still have to understand the ccxt api a little better to maybe come up with a more fitting solution for this, so that this could be standardized for all supported exchanges even those like kraken which do not use / as a seperator

Reply:
This code is obsolete and is now entirely handled by graph .py